### PR TITLE
Create Label Component

### PR DESCRIPTION
**Pull request by code-sanchu**
Created label component. Followed project code style. Implemented tests.

WIP: Couldn't quite get a good test for checking if label text isn't selected on double click due to a possible bug in Firefox. In short: in Firefox, for the user double-clicking the label works as intended; but, in JS, `window.getSelection()` returns label text as selected even when it isn't for the user. I opened a stack-overflow issue which is currently in 'staging ground' because I just created a new Stack Overflow account, but can provide link later.

### CODE REVIEW

File: src/lib/components/Label/tests/LabelTest.svelte [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<script lang="ts">
+	import { Label } from '$lib/components/Label';
+	import { labelDataTestId, inputId, labelText } from './constants';
+
+	type MouseDownEvent = MouseEvent & {
+		currentTarget: EventTarget & HTMLLabelElement;
+	};
+	export let onMouseDown: (event: MouseDownEvent) => void;
```
Comment (TGlide) [inline src/lib/components/Label/tests/LabelTest.svelte:8]: This sin't being used, right? If so, you can delete it

File: src/lib/components/Label/tests/constants.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export const labelDataTestId = 'label';
```
Comment (TGlide) [inline src/lib/components/Label/tests/constants.ts:1]: You can put these directly in the test files, no need to centralize them. Alternatively, export them from `LabelTest.svelte` by adding a `module` script tag: https://svelte.dev/docs#component-format-script-context-module

**Review by TGlide — CHANGES_REQUESTED** [review]
Thank your for the PR!

Would you be able to create a preview file too for the docs? It would be greatly appreciated 😄 But don't worry if you do not have the time.

File: src/lib/components/Label/tests/LabelTest.svelte (hunk 1 above)
Comment (code-sanchu) [reply src/lib/components/Label/tests/LabelTest.svelte:8]: You're right. I'll remove.

**Comment by code-sanchu** [comment]
> Would you be able to create a preview file too for the docs? It would be greatly appreciated smile But don't worry if you do not have the time.

Yes no probs. I'll do this :)

File: src/lib/components/Label/tests/constants.ts (hunk 2 above)
Comment (code-sanchu) [reply src/lib/components/Label/tests/constants.ts:1]: Done.

**Review by TGlide — APPROVED** [review]

**Comment by TGlide** [comment]
Thank you @code-sanchu !!

**Pull request merged**